### PR TITLE
Use optionals in parsing command line arguments.

### PR DESCRIPTION
**Pull request by grp**
Instead of checking if strings are empty, use optionals to store if
the arguments were passed at all. Additionally, add a few utility
methods to the option parser to simplify various tool option classes.

### CODE REVIEW

**Comment by facebook-github-bot-0** [comment]
Thank you for your pull request.  As you may know, we require contributors to sign our Contributor License Agreement, and we don't seem to have you on file and listed as active anymore.  In order for us to review and merge your code, please email cla@fb.com with your details so we can update your status.

**Comment by modocache** [comment]
@grp It looks like your previous CLA is no longer "active". I think you should be able to renew your agreement (and shut these bots up) by visiting https://code.facebook.com/cla.

**Comment by grp** [comment]
I sent a new one to @caabernathy last week (the online form errors out), assuming it just hasn't been added into the system yet.

**Comment by modocache** [comment]
Awesome, thanks! Sorry for the noise :)

**Comment by grp** [comment]
@Ktwu, not sure why, but your name doesn't seem to load in the assignees dropdown. Maybe you need to join the @facebook org?

File: Libraries/libutil/Sources/Options.cpp [hunk 1]
```diff
@@ -31,42 +34,93 @@ NextString(std::string *result, std::vector<std::string> const &args, std::vecto
...
-        if (str == "YES") {
-            *result = true;
-        } else if (str == "NO") {
-            *result = false;
+    std::string const &arg = **it;
+    if (++*it != args.end()) {
+        std::string const &value = **it;
+
+        if (!*result || allowDuplicate) {
+            if (value == "YES") {
```
Comment (marcsalem) [inline Libraries/libutil/Sources/Options.cpp]: This looks like "no" and "yes" will fail. Is it really expected to be case sensitive?

**Comment by marcsalem** [comment]
Looks good to me.

**Comment by Ktwu** [comment]
@grp, I've added myself to the @facebook org, so hopefully I show up in the future.

**Pull request merged**